framework/src/main/java/org/checkerframework/framework/util/StringToJavaExpression.java

```diff
@@ -0,0 +1,362 @@
+            throws JavaExpressionParseException {
+        ExecutableElement ee = TreeUtils.elementFromUse(newClassTree);
+        JavaExpression javaExpr = StringToJavaExpression.atMethodDecl(expression, ee, checker);
+        return javaExpr.atConstructorInvocation(newClassTree);
+    }
+
+    /**
+     * Parses a string as if it were written at the declaration the field and then viewpoint-adapts
+     * the result to the use.
+     *
+     * @param expression a Java expression
+     * @param fieldAccess the field access tree
+     * @param checker checker used to get the {@link
+     *     javax.annotation.processing.ProcessingEnvironment} and current {@link
+     *     com.sun.source.tree.CompilationUnitTree}
+     * @return a {@code JavaExpression} for {@code expression}
+     * @throws JavaExpressionParseException if {@code expression} cannot be parsed
+     */
+    static JavaExpression atFieldAccess(
+            String expression, MemberSelectTree fieldAccess, SourceChecker checker)
+            throws JavaExpressionParseException {
+
+        Element ele = TreeUtils.elementFromUse(fieldAccess);
+        if (ele.getKind() != ElementKind.FIELD && ele.getKind() != ElementKind.ENUM_CONSTANT) {
+            throw new BugInCF("Expected a field, but found %s.", ele.getKind());
+        }
+        VariableElement fieldEle = (VariableElement) ele;
+        JavaExpression receiver = JavaExpression.fromTree(fieldAccess.getExpression());
+        JavaExpression javaExpr = StringToJavaExpression.atFieldDecl(expression, fieldEle, checker);
+        return javaExpr.atFieldAccess(receiver);
+    }
+
+    /**
+     * Parses a string as if it were written at the one of the parameters of {@code lambdaTree}.
+     * Parameters of the lambda are expressed as {@link LocalVariable}s.
+     *
+     * @param expression a Java expression
+     * @param lambdaTree the lambda tree
+     * @param parentPath path to the parent of {@code lambdaTree}; required because the expression
+     *     can reference final local variables of the enclosing method
+     * @param checker checker used to get the {@link
+     *     javax.annotation.processing.ProcessingEnvironment} and current {@link
+     *     com.sun.source.tree.CompilationUnitTree}
+     * @return a {@code JavaExpression} for {@code expression}
+     * @throws JavaExpressionParseException if {@code expression} cannot be parsed
+     */
+    static JavaExpression atLambdaParameter(
+            String expression,
+            LambdaExpressionTree lambdaTree,
+            TreePath parentPath,
+            SourceChecker checker)
+            throws JavaExpressionParseException {
+
+        TypeMirror enclosingType = TreeUtils.typeOf(TreePathUtil.enclosingClass(parentPath));
+        JavaExpression receiver = JavaExpression.getPseudoReceiver(parentPath, enclosingType);
+        ThisReference thisReference =
+                receiver instanceof ClassName ? null : (ThisReference) receiver;
```

Comment:
I'm confused by this line.  Should the instance of "ClassName" be "ThisReference"?